src/components/Premium/PremiumBrand.tsx

```diff
@@ -1,32 +1,104 @@
+import { useEffect, useState } from 'react';
 import styled from 'styled-components';
 
-import { BrandNameListIcon } from '../../assets/icon';
-import { 꼬달리Img, 다비네스Img, 달팜Img, 더바디샵Img } from '../../assets/image';
+import { BrandSortIcon, BrandSortReverseIcon } from '../../assets/icon';
+import { BrandData } from '../../types/brand';
+import { getBrandData } from '../../utils/lib/brand';
+
+import PremiumBrandCard from './PremiumBrandCard';
 
 const PremiumBrand = () => {
+  
+  // 1. API 연결하여 브랜드 리스트 받아오기
+  
+  const [PremiumBrandList, setBrandList] = useState<BrandData[]>([]); 
+
+  const getBrandList = async () => {
+    try {
+      const { data: { data } } = await getBrandData();
+      setBrandList(data);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  useEffect(() => {
+    getBrandList();
+  }, []);
+  
+  // 2. 정렬 버튼 누르면 정렬
+
+  const [isAscending, setIsAscending] = useState<boolean>(true);
+  
+  const SortBrandList = () => {
+    const sortedList = [...PremiumBrandList].sort((a, b) => {
+      const nameA = a.name.toLowerCase();
+      const nameB = b.name.toLowerCase();
+
+      if (isAscending) {
+        if (nameA < nameB) return -1;
+        if (nameA > nameB) return 1;
+      } else {
+        if (nameA > nameB) return -1;
+        if (nameA < nameB) return 1;
+      }
+
+      return 0;
+    });
```

Comment:
```suggestion
    const sortedList = PremiumBrandList.sort((a, b) => {
      const nameA = a.name.toLowerCase();
      const nameB = b.name.toLowerCase();
      
      return isAscending? nameA-nameB : nameB-nameA;
```

아마 이렇게 간단히 할 수 있을고야!

Reply:
이거머야 미쳤다 정말 똑똑하잔아... 고마워!!

Reply:
문자열이라 뺄셈은 안된다구 해서 찾아보니 localeCompare 로 비교가 가능하구먼!
alert('a'.localeCompare('b')); // -1
alert('b'.localeCompare('a')); // 1
alert('b'.localeCompare('b')); // 0

Reply:
``` return isAscending ? nameA.localeCompare(nameB) : nameB.localeCompare(nameA); ``` 요렇게